packages/core/src/browser/quick-input/quick-command-service.ts

```diff
@@ -0,0 +1,156 @@
+/********************************************************************************
+ * Copyright (c) 2021 SAP SE or an SAP affiliate company and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { inject, injectable } from 'inversify';
+import { Disposable, Command, CommandRegistry } from '../../common';
+import { ContextKeyService } from '../context-key-service';
+import { CorePreferences } from '../core-preferences';
+import { QuickInputContribution } from './quick-input-contribution';
+import { QuickInputService } from './quick-input-service';
+
+export const quickCommand: Command = {
+    id: 'workbench.action.showCommands'
+};
+
+export const CLEAR_COMMAND_HISTORY: Command = {
+    id: 'clear.command.history',
+    label: 'Clear Command History'
+};
+
+@injectable()
+export class QuickCommandService extends QuickInputService implements QuickInputContribution {
```

Comment:
I don't understand why this service should extend QuickInputService. All we're interested in is a the `QuickInputContribution`.

Reply:
The thought behind it is:

1. Base class = `QuickInputService` which is rebinded to `MonacoQuickInputService`.
2. `QuickCommandService`, `QuickEditorService` etc. are all derivatives of `QuickInputService` and are also rebinded to their relevant implemented class (i.e. `MonacoQuickCommandService`)
3. Now, when you want to open commands quickInput widget - you want to invoke "open" method on `MonacoQuickInputService` with relevant prefix = '>' which `MonacoQuickCommandService` is essentially extending.

This all is based on previous implementation design - QuickOpenService > MonacoQuickOpenService. I do not want to create multiple instances of `MonacoQuickInputService` but have one "access point" to `QuickInput` functionality through MQIS.

Alternatively, I could see if I can remove all `core` services (quickCommand/editor/helper) and just leave `QuickInputService`. What do you think?

Reply:
> Alternatively, I could see if I can remove all core services (quickCommand/editor/helper) and just leave QuickInputService. What do you think?

On second thought, I'm not sure it is possible as I need in `core` to run "registerQuickAccessProvider" but for command, I want it to register command QuickAccessProvider.. how would you do that in another way?